src/Files.App/Utils/Shell/ShellFileOperations2.cs

```diff
@@ -0,0 +1,744 @@
+﻿using System.Runtime.InteropServices;
+using Vanara.PInvoke;
+using static Vanara.PInvoke.Shell32;
+using static Vanara.Windows.Shell.ShellFileOperations;
+
+namespace Vanara.Windows.Shell;
+
+/// <summary>Queued and static file operations using the Shell.</summary>
+/// <seealso cref="IDisposable"/>
+/// https://github.com/dahall/Vanara/blob/master/Windows.Shell.Common/ShellFileOperations/ShellFileOperations.cs
+public class ShellFileOperations2 : IDisposable
+{
+	private const OperationFlags defaultOptions = OperationFlags.AllowUndo | OperationFlags.NoConfirmMkDir;
+	private bool disposedValue = false;
+	private IFileOperation op;
+	private OperationFlags opFlags = defaultOptions;
+	private HWND owner;
+	private readonly IFileOperationProgressSink sink;
+	private readonly uint sinkCookie;
+
+	/// <summary>Initializes a new instance of the <see cref="ShellFileOperations"/> class.</summary>
+	/// <param name="owner">The window that owns the modal dialog. This value can be <see langword="null"/>.</param>
+	public ShellFileOperations2(HWND owner = default)
+	{
+		Ole32.CoCreateInstance(typeof(CFileOperations).GUID, null, Ole32.CLSCTX.CLSCTX_LOCAL_SERVER, typeof(IFileOperation).GUID, out var opObj);
```

Comment:
The trick is using CLSCTX_LOCAL_SERVER, CLSCTX_INPROC_SERVER does not seem to work, meh.